primary/src/garbage_collector.rs

```diff
@@ -57,9 +61,19 @@ impl<PublicKey: VerifyingKey> GarbageCollector<PublicKey> {
     async fn run(&mut self) {
         let mut last_committed_round = 0;
         while let Some(certificate) = self.rx_consensus.recv().await {
-            // TODO [issue #9]: Re-include batch digests that have not been sequenced into our next block.
-
             let round = certificate.round();
+
+            // Re-include batch digests that have not been sequenced into our next block.
+            if certificate.author() == name {
+                for payload in certificate.header.payload {
+                    self.tx_our_digests
```

Comment:
Probably this might be too much to change, but do we want to introduce some concept of "maximum" retries to avoid some "infinite" re-inclusions? I guess the probability might be small for this to happen (I have no data to support the argument, it's pure finger in the air estimation) but we might want to have maximum number of retries for the digests which could be piggy backed to the digests circulated within the app. When max retries to be included reached then we completely discard with related log and metrics.